### PR TITLE
UPSTREAM: 63206: Partial pick of restmapper fixes

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1577520

Partially picks ef0d1ab81927214db80c30d5af491f67546d790b from upstream PR https://github.com/kubernetes/kubernetes/pull/63206 to prevent squashing url errors when attempting to discover a resource list from the server.

Partially picks 5432ef5c4557fc47c97f631ac3b23738bbd27115 from upstream PR https://github.com/kubernetes/kubernetes/pull/63250 to prevent squashing url errors when attempting to find a REST mapping for a resource.

**Before**
```
# with no connection to a cluster
$ oc get dc
error: the server doesn't have a resource type "dc"
```

**After**
```
# with no connection to a cluster
$ oc get dc
The connection to the server 127.0.0.1:8443 was refused - did you specify the right host or port?
```

cc @smarterclayton @deads2k @soltysh

### CODE REVIEW

**Comment by soltysh** [comment]
Get this green and merge.

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go [hunk 1]
```diff
@@ -89,8 +89,7 @@ func (f *ring1Factory) objectLoader() (meta.RESTMapper, runtime.ObjectTyper, err
 		groupResources, err = discovery.GetAPIGroupResources(discoveryClient)
 	}
 	if err != nil {
-		glog.V(3).Infof("Unable to retrieve API resources, falling back to hardcoded types: %v", err)
-		return legacyscheme.Registry.RESTMapper(), legacyscheme.Scheme, nil
```
Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go]: @soltysh this breaks `--local` for `oc` commands

Comment (soltysh) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go]: Try figuring out why, it's always problematic when doing partial picks. Maybe you're missing something from upstream. Try debugging the path when `--local` is invoked in origin and see what might fix it.

**Comment by soltysh** [comment]
It looks like those partial cherry-picks are not sufficient, but instead require more picks to be able to set the scheme when working with `--local` flag. I'm worried that these will lead to more instability in the CLI than it's worth the cost. My suggestion is to leave it as is for 3.10 and fix it in 3.11. /cc @smarterclayton since you asked for it in the original BZ.

**Comment by soltysh** [comment]
This should be picked by the rebase.
/close

**Pull request closed without merging**